app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,76 @@
+<div class="recipe-detailed">
+<p style="color: green"><%= notice %></p>
+<div class="recipe-show"> 
+  <%= render @recipe %>
+  <div class="link-btn-recipe">
+    <button class="generate-shopping-list-btn"> Generate Shopping list </button>
+    <%= button_to "Add Ingredients" %>
+  </div>
+
+  <table id="t-recipe-details">
+    <thead>
+      <tr>
+        <th>Food</th>
+        <th>Quantity</th>
+        <th>Value</th>
+        <th>Actions</th>
+      </tr>
+    </thead>
+    <tbody>
+
+        <% RecipeFood.includes([:food]).where(recipe_id: @recipe.id).each do |recipe_food| %>
+          <tr>
+          <td><%= Food.find_by(id: recipe_food.food_id).name %></td>
+          <td> <%= recipe_food.quantity %> </td>
+          <td> $ <%= recipe_food.quantity.to_i * recipe_food.food.price %> </td>
+          <td>
+          <%= button_to "Modify", recipe_food, method: :edit, data: { confirm: "Are you sure?" } %>
+          <%= button_to "Remove", recipe_food, method: :delete, data: { confirm: "Are you sure?" } %>
+          </td>
+          </tr>
+        <% end %>
+
+    </tbody>
+  </table>
+  <br/>
+  <div class="add_recipe_food_form">
+  <%= form_with(model: @recipe_food) do |form| %>
+    <% if @recipe_food.errors.any? %>
+      <div style="color: red">
+          <h2><%= pluralize(@recipe_food.errors.count, "error") %> prohibited this recipe_food from being saved:</h2>
+
+          <ul>
+            <% @recipe_food.errors.each do |error| %>
+              <li><%= error.full_message %></li>
+            <% end %>
+          </ul>
+        </div>
+    <% end %>
+        <div class="food_collection_recipe">
+          <%= collection_select(:recipe_food, :food_id, Food.all, :id, :name) %>
+          <%= form.label :quantity, style: "display: block" %>
+          <%= form.number_field :quantity %>
+          <%= form.hidden_field :recipe_id, value: params[:id] %>
+          <div>
+            <%= form.submit id: "create_recipe_food_btn" %>
+          </div>
+        </div>
+      </div>
+  <% end %>
+
+  <div class="generate-shopping-list-container" id="pop-up">
+      <%=form_with url: general_shopping_list_index_path, method: :get do |f| %>
+        <div>
```

Comment:
- [ ] Please kindly ensure that this form only appears when the "Add ingredient" button is clicked.

![image](https://user-images.githubusercontent.com/43995830/219483467-eaa5b7ae-857e-41e2-b7e7-5d1094115508.png)